### PR TITLE
Update Firebase JWT to version 6.x

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This aims to Upgrade Firebase JWT package to 6.x version

Before
----------------------------------------
Firebase JWT Package 5.x version used

After
----------------------------------------
Firebase JWT Package 6.x version used

I believe this will have fixed the test failures that the previous dependabot PR hit

ping @totten

### CODE REVIEW

**Comment by seamuslee001** [comment]
Cross linking to the previous dependabot PR https://github.com/civicrm/civicrm-core/pull/25603

File: composer.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "php": "~7.3 || ~8",
     "composer-runtime-api": "~2.0",
     "dompdf/dompdf" : "~2.0.2",
-    "firebase/php-jwt": ">=3 <6",
+    "firebase/php-jwt": ">=3 <7",
```
Comment (totten) [inline composer.json:56]: @seamuslee001 Ah, cool that this change looks as small as it does. 

The only thing is... I don't think `Firebase\JWT\Key` exists in all versions `>=3 <7`. (Skimming https://github.com/firebase/php-jwt/tree/v5.0.0/src, it looks like it was added somewhere during 5.x. But this constraint does reflect something else important - that the version-constraints are externally-influenced. (Just as we can't keep everyone on 5.x for as long as we please... we probably can't get everyone to 6.x when we please.)

In theory, if we were strictly supporting 6.x, then I think the loops/arrays could be simplified a lot. But there is another possibility -- take the approach you have above (2 changes; circa lines 76 and 82) and make those changes conditional:

```php
$useKeyObj = version_compare(\Composer\InstalledVersions::getVersion('firebase/php-jwt'), '6', '>=');
if (!$useKeyObj) Civi::log()->info("Using deprecated version of firebase/php-jwt. Upgrade to 6.x+.");
```
```php
$keysByAlg[$alg][$key['id']] = ($useKeyObj) 
  ? new Key($key['key'], $alg) 
  : $key['key'];
```
```php
return $useKeyObj 
  ?  (array) JWT::decode($token, $keys)
  : (array) JWT::decode($token, $keys, [$alg]);
```

Comment (seamuslee001) [reply composer.json:56]: @totten ok I have added in that compatibility layer now

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by totten** [comment]
On [MM](https://chat.civicrm.org/civicrm/pl/c1tr1c1ggbbf9fj7zd4k8gbxre), Seamus mentioned that he's been doing some additional testing to confirm that it works with the older `php-jwt`. (Log: https://gist.github.com/seamuslee001/3b3f67656d80a73554978956d40bc049)

If this passes that testing and the CI testing, then I think we're good to merge.

**Comment by seamuslee001** [comment]
Test fail looks environmental

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
Merging based on @totten's comment above

**Pull request merged**